api/src/infrastructure/docker.rs

```diff
@@ -1346,4 +1475,121 @@ mod tests {
             )
         );
     }
+
+    #[tokio::test]
+    async fn should_create_volume_mount_for_new_deployment() -> Result<(), ShipLiftError> {
```

Comment:
I'm not sure about these tests because it is require to a running Docker backend which may pollute your host system if any fails. For integration testing we have a separate crate in the project. If you want, we can discuss in a video chat.